alternator/executor.cc

```diff
@@ -1861,11 +1864,57 @@ std::unordered_set<std::string> calculate_attrs_to_get(const rjson::value& req,
     return {};
 }
 
-static std::optional<rjson::value> describe_single_item(schema_ptr schema,
+/**
+ * Helper routine to extract data when we already have 
+ * row, etc etc.
+ * 
+ * Note: include_all_embedded_attributes means we should
+ * include all values in the `ATTRS_COLUMN_NAME` map column.
+ * 
+ * We could change the behaviour to simply include all values 
+ * from this column if the `ATTRS_COLUMN_NAME` is explicit in 
+ * `attrs_to_get`, but I am scared to do that now in case 
+ * there is some corner case in existing code.
```

Comment:
This comment is based on a misunderstanding on what attrs_to_get_is... It is a list of DynamoDB attributes. The "ATTRS_COLUMN_NAME" is just an implmentation detail in Alternator. 
I think this function would have been better off without this comment at all which casts doubts more than it explains. The name "include_all_embedded_attribute" is quite meaningless, there is no notion of "embedded attributes" - the same attribute may be in the map (ATTRS_COLUMN_NAME) or as a separate column if it is a key of the base table or LSI f the table - it's not clear why we need a separate option to return the first type of attributes but not the second - and the name "embedded" doesn't quire explain the distinction.

Reply:
As explained in the code: The problem with the attributes in the :attr column is that they are not visible in the scylla schema. I want to get all data without having to figure out what data is called in the map (the ones in columns we know already, implicitly). I am open to modifying the API here, but the basic requirement remains, and I felt this was least resistance.

Reply:
If you want to get *all* columns you could have called the flag that - no need to list separately the columns in the schema and another flag for those not in the schema (which you called "embedded"). But heck, I think we already return all the columns if the list of required attributes is empty, didn't we???
I'll fix in a followup patch, if I can (maybe as I try to fix it I'll find I'm misunderstanding something?)

Reply:
But I don't explicitly want _all_ columns. I want those mentioned and those embedded in :attr. The problem with empty attr_map is that it then don't return any _actual_ column values. So if we have mixed real column/embedded values we would have to call twice. And merge.